internal/ingress/annotations/redirect/redirect.go

```diff
@@ -73,14 +74,28 @@ func (a redirect) Parse(ing *extensions.Ingress) (interface{}, error) {
 		return nil, err
 	}
 
+	prc, err := parser.GetStringAnnotation("permanent-redirect-code", ing)
+	if err != nil && !errors.IsMissingAnnotations(err) {
+		return nil, err
+	}
+
+	var prci = http.StatusPermanentRedirect
+	if prc != "" {
```

Comment:
If you do use an `int`, the validation of the annotation could be something like
```golang
const defaultPermanentRedirectCode = http.StatusPermanentRedirect

if prc < http.StatusMultipleChoices || prc > http.StatusPermanentRedirect {
prc = defaultPermanentRedirectCode
}
```
(depending on what HTTP codes you believe are appropriate).

The const just makes the code mode readable, but more importantly you don't need 2 variables anymore.